pkg/filter/filter.go

```diff
@@ -108,24 +119,41 @@ func JoinResourceTypeMatchers(ms ...ResourceTypeMatcher) ResourceTypeMatcher {
 
 // ResourceRequirement allows specifying a resource requirement by type and/or name
 type ResourceRequirement struct {
+	// Provides the Name of the resource object
 	v1.LocalObjectReference `json:",inline,omitempty"`
+	// Provides the Group, Version, and Resource values (GVR)
 	ResourceTypeRequirement `json:",inline,omitempty"`
+	// Specifies a set of label requirements to be used as filters for matches
+	metav1.LabelSelector `json:",inline,omitempty"`
 }
 
-// Matches returns true if the specified resource name/GVR matches the requirement
-func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource) bool {
+// Matches returns true if the specified resource name/GVR/labels matches the requirement
+func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource, resourceLabels map[string]string) bool {
 	// If the requirement does not specify a resource name - only check the
 	// ResourceTypeRequirement i.e. GVR match
-	if r.LocalObjectReference.Name == "" {
-		return r.ResourceTypeRequirement.Matches(gvr)
+	if !(matches(r.Name, name) && r.ResourceTypeRequirement.Matches(gvr)) {
+		return false
 	}
-	return matches(r.Name, name) && r.ResourceTypeRequirement.Matches(gvr)
+	if len(resourceLabels) == 0 || (len(r.MatchExpressions) == 0 && len(r.MatchLabels) == 0) {
+		// If there are no resourceLabels to be matched, the LabelSelector is ignored.
```

Comment:
Thanks for the comments and clearly stated rationale.  I don't think this is the policy that we want.  Consider the use case of "match everything with label foo=bar".

Reply:
This would occur if the calling code failed to pass in the resource labels. Would it be better to return an error if LabelSelector is specified and no labels are supplied? Or do we want to assume that no match in that case occurs? I think we have the logic working. Now we need to decide what the policy would be. I think you're right but I want to think through each case and gather input. @vkamra @tdmanv ??

Reply:
@tylerje I'm in agreement ... time to write up what the inputs and outputs would be of various scenarios.  Overall you are on the right track.

Reply:
Two tests added. This required a change to the logic.